### PR TITLE
test: split input/output for planner test

### PR DESCRIPTION
**Pull request by xxchan**
I hereby agree to the terms of the [RisingWave Labs, Inc. Contributor License Agreement](https://gist.github.com/TennyZhuang/f00be7f16996ea48effb049aa7be4d66#file-rw_cla).

## What's changed and what's your intention?

So that we can edit input files painlessly (e.g., no merge conflicts). The output files still contain the inputs for readability, and have exactly the same format as before. 

(For reviewer: to confirm output is unchanged, run `git diff --stat` locally. GitHub is too slow 🤪)

<details><summary>If there are no such changes, it should be good</summary>
<p>

image

</p>
</details> 

close https://github.com/risingwavelabs/risingwave/issues/8557

The trick is similar to #9787, i.e., make planner tests snapshot testing.

Note: The new result check is simply assert eq the whole file, and produce the diff, instead check test cases one by one.

## Limitations

- In theory the code can be further refactored, but I only made minimal changes to make it work.
- The error message produced by `expect-test` isn't optimal (see https://github.com/rust-analyzer/expect-test/issues/39). But since the workflow is mostly just run `dapt`, and then use external diff tool to review it, it might be acceptable. Alternatively we can fork and improve `expecte-test`, or use `insta` instead.

## Documentation

- [x] My PR **DOES NOT** contain user-facing changes.

<details><summary>Click here for Documentation</summary>

### Types of user-facing changes

Please keep the types that apply to your changes, and remove the others.

- Installation and deployment
- Connector (sources & sinks)
- SQL commands, functions, and operators
- RisingWave cluster configuration changes
- Other (please specify in the release note below)

### Release note

</details>

### CODE REVIEW

File: src/frontend/planner_test/tests/testdata/input/AAA_I_AM_A_NEW_TEST.yaml
Comment (xxchan) [inline src/frontend/planner_test/tests/testdata/input/AAA_I_AM_A_NEW_TEST.yaml:1]: This is for reviewer to play with only

**Review by chenzl25 — APPROVED** [review]
Generally LGTM. Maybe we need to tell developers that `apply-planner-test` no longer works and we only need `do-apply-planner-test`.

**Review by st1page — APPROVED** [review]
LGTM! So thanks.

**Comment by st1page** [comment]
just to confirm, what will happen when some cases panicked? And how a developer can not which case panic in the new framework? In other words, we might need some logic to replace this code.

https://github.com/risingwavelabs/risingwave/blob/68140f3a0b8236516f7b75c8b681985003b9b33d/src/frontend/planner_test/src/bin/apply.rs#L26-L39

**Review by TennyZhuang — APPROVED** [review]
LGTM

**Review by TennyZhuang — APPROVED** [review]
LGTM

**Comment by xxchan** [comment]
> just to confirm, what will happen when some cases panicked? And how a developer can not which case panic in the new framework? In other words, we might need some logic to replace this code.

We don't need to handle that now. We used to do it because `apply` is a separate binary. And it tells you to run `run-planner-test` (`nextest`). Now we only run `nextest` which catches panic for us.

**Review by stdrc — COMMENTED** [review]
LGTM

**Pull request merged**